Allow control of cleanup timing

This library sets shutdown hooks. This can lead to some very strange class loading problems in Maven 3.2.x and above, as shown below. As best I can work out, the only fix here is to expose an API to allow a maven plugin that is calling this library (or a library that calls this library, like the mongo embedded) to explicitly shut things down when it's done.

```
Exception in thread "Thread-6" java.lang.NoClassDefFoundError: org/apache/commons/io/FileUtils
    at de.flapdoodle.embed.process.io.file.Files.forceDelete(Files.java:104)
    at de.flapdoodle.embed.process.runtime.AbstractProcess.stop(AbstractProcess.java:168)
    at de.flapdoodle.embed.process.runtime.Executable.stop(Executable.java:71)
    at de.flapdoodle.embed.process.runtime.Executable$JobKiller.run(Executable.java:88)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.io.FileUtils
    at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
    at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
```

--- a/src/main/java/de/flapdoodle/embed/process/io/file/Files.java
+++ b/src/main/java/de/flapdoodle/embed/process/io/file/Files.java
@@ -43,6 +43,11 @@ public class Files {
 
 	private static Logger logger = LoggerFactory.getLogger(Files.class);
 	public static final int BYTE_BUFFER_LENGTH = 1024 * 16;
+	/**
+	 * Instance to force loading {@link DeleteDirVisitor} class to avoid
+	 * {@link NoClassDefFoundError} in shutdown hook.
+	 */
+	private static final SimpleFileVisitor<Path> DELETE_DIR_VISITOR = new DeleteDirVisitor();
 
 	private Files() {
 
@@ -149,10 +154,8 @@ public class Files {
 	}
 
 	private static class DeleteDirVisitor extends SimpleFileVisitor<Path> {
-		private static final SimpleFileVisitor<Path> instance = new DeleteDirVisitor();
-
 		public static SimpleFileVisitor<Path> getInstance() {
-			return instance;
+			return DELETE_DIR_VISITOR;
 		}
 
 		@Override

